### ISSUE TITLE
A quick question about computing the metrics for genes and cells

### ISSUE THREAD

**Issue by HelloWorldLTY** [GSK-AI/seq2cells#3]
Hi, thanks for your great work. After reviewing your codes to compute the correlation, I have some questions.

```
def get_across_x_correlation(
    ad: ad.AnnData,
    axis: Literal[0, 1] = 0,
    obs_layer: str = "X",
    pred_layer: str = "predicted",
    return_per_entry: bool = False,
):
    """Calculate correlation pred vs obs across given axis

    Arguments
    ---------
    ad: ad.AnnData
        AnnDataFrame hosting observations,
        should be in the orientation gene(obs) x cells(var).
    axis: int [0,1]
        Select the axis across which to correlate
        observations with predictions.
        0 - across genes; 1 - across cells
    obs_layer: str = 'X'
        Name of the anndata layer in which the observations are
        stored. If 'X' is supplied will use the default anndata.X
        matrix. Default = 'X'
    pred_layer: str = 'predicted'
        Name of the anndata layer in which the observations are
        stored. If 'X' is supplied will use the default anndata.X
        matrix. Default = 'predicted'
    return_per_entry: bool = False
        Toggle if to return a dataframe with the per entry correlations.

    Output
    ------
    mean_cor: float
        The mean Pearson correlation across all entries.
    cor_per_entry: [Optional] pd.DataFrame
        Listing the correlation per entry along the non selected axis.

    Notes
    -----
    Will compute the correlation of observations vs. predictions
    per entry. Depending on the axis selected:
    axis=0 -> across genes -> entries are cells
    axis=1 -> across cell (types) -> entries are cells
    """
    print(ad.shape)
    if obs_layer == "X":
        obs = ad.to_df()
    else:
        obs = ad.to_df(layer=obs_layer)

    if pred_layer == "X":
        pred = ad.to_df()
    else:
        pred = ad.to_df(layer=pred_layer)

    # correlation across genes per cell
    cor_per_entry = obs.corrwith(pred, axis=axis)

    mean_cor = np.nanmean(cor_per_entry)

    if return_per_entry:
        return mean_cor, cor_per_entry
    else:
        return mean_cor

```

1. It seems that your comments mention that "   axis=1 -> across cell (types) -> entries are cells" I think there exists a typo. It should be entries are genes, otherwise the commetns for axis=0 and axis=1 are the same.
2. I check the shape of anndata for computing the metrics, which are strange to me:
no hvg:
(11824, 2600)
hvg:
(5000, 2600)

Should we only use the genes for testing to compute the metrics? If so, I found that I receive high low across gene correlation and high cell correlation. Does this conclusion conflict with your figure 3? Thanks a lot.

**Comment by rschwess** [comment]
Hi HelloWorldLTY,

1. Yes thats a typo it should read `entries are genes`. Also the comment above the correlation computation only covers one use case. 

2. So we transposed the AnnDataFrame to be genes x cells, because each gene is a training example for seq2cells, but I acknowledge this is a bit counter intuitive to start. 
 
nohvg --> 11_824 genes and 2_600 cells 
subsetting to hvg genes --> 5_000 genes and 2_600 cells 

For the results in figure 3 we only used test set genes, although the data in the paper are from the HSC dataset and you are looking at the PBMC example. 

We did however NOT subset genes to only highly variable genes. We trained and evaluated on all genes (after train, val, test split).

When you calculate the correlations for only the highly variable genes the values will be different to what you see in Figure 3. I believe in the hvg case for the HSC dataset we saw something similar to what you observed for the PBMC dataset:

* higher across cell correlation --> relatively more variability to learn when ditching the genes that are more or less the same in all cells, and learning the variation in those later is probably hard because it's mostly noise.

* lower cross gene correlation --> focussing on the hvg genes, esp. when called liberally as we did also contains many very noisy genes while you drop many genes that are readily detected, just not variable enough across cells.

For the paper we chose to stick to a single evaluation on all genes, but one could stratify the correlation over highly expressed genes, highly variable genes and also the conservativeness with which they are called. 

Hope this helps
